File: BOJ/Recommended1/Bronze/17945/17945.swift
let input = readLine()!.split(separator: " ").map{Int(String($0))!}
let a = input[0]
let b = input[1]
let bMin = min(b,-b)
let bMax = max(b,-b)
if a != 0 && b == 0 {print(-2*a,terminator: " ")}
for x in bMin...bMax {
    if x * (x+2*a) == -b {print(x,terminator: " ")}
}

File: BOJ/Recommended1/Bronze/17945/17945.py
a,b = map(int,input().split())
_max = max(b,-b)
_min = min(b,-b)
ans = ""
if a != 0 and b == 0 :
    ans+= f"{-2*a} "
for x in range(_min,_max+1) :
    if x * (x+2*a) == -b :
        ans+= f"{x} "
print(ans)